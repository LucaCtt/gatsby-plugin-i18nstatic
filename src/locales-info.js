import { createContext, useContext } from "react";

export const LocalesInfoContext = createContext({
  currentLocale: "",
  defaultLocale: "",
  locales: []
});

export const useLocalesInfo = () => {
  const { currentLocale, defaultLocale, locales } = useContext(
    LocalesInfoContext
  );
  return [currentLocale, defaultLocale, locales];
};
