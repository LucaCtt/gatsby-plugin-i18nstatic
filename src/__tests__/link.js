import React from "react";
import { render } from "react-testing-library";

import Link from "../link";
import { useLocalesInfo } from "../locales-info";

jest.mock("../locales-info", () => ({
  useLocalesInfo: jest.fn(() => ["it", "en", ["it", "en", "es"]])
}));

const TestComponent = ({ children, ...props }) => (
  <Link to="/" {...props}>
    {children}
  </Link>
);

test("Renders correctly", () => {
  const { container } = render(<TestComponent>test</TestComponent>);
  expect(container.firstChild).toMatchSnapshot();
});

test("Locale passed as prop overrides default locale", () => {
  const { container } = render(<TestComponent locale="es">test</TestComponent>);
  expect(container.firstChild).toMatchSnapshot();
});

test("Handles default locale correctly", () => {
  useLocalesInfo.mockImplementation(() => ["en", "en", ["en", "it"]]);
  const { container } = render(<TestComponent>test</TestComponent>);
  expect(container.firstChild).toMatchSnapshot();
});
