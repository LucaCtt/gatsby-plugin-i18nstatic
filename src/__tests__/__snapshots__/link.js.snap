// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handles default locale correctly 1`] = `
<div
  data-testid="gatsby-link"
  to="/"
>
  test
</div>
`;

exports[`Locale passed as prop overrides default locale 1`] = `
<div
  data-testid="gatsby-link"
  to="/es/"
>
  test
</div>
`;

exports[`Renders correctly 1`] = `
<div
  data-testid="gatsby-link"
  to="/it/"
>
  test
</div>
`;
